src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateExpiryMonitor.java

```diff
@@ -92,7 +95,7 @@ void watchForCertExpiryOnce() {
      * @param cg certificate generator
      */
     public void addToMonitor(CertificateGenerator cg) {
-        monitoredCertificateGenerators.add(cg);
+        certificateGeneratorRegistry.registerGenerator(cg);
```

Comment:
Why didn't I remove this `addToMonitor` methods? - Simple answer is to avoid changing yet more files, I would have had to update some tests and this PR had already a decent amount of changes.